object-api-disk-not-found_test.go

```diff
@@ -0,0 +1,264 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+)
+
+// Wrapper for calling TestListObjectPartsDiskNotFound tests for both XL multiple disks and single node setup.
+func TestListObjectPartsDiskNotFound(t *testing.T) {
+	ExecObjectLayerDiskNotFoundTest(t, testListObjectPartsDiskNotFound)
+}
+
+// testListObjectParts - Tests validate listing of object parts when disks go offline.
+func testListObjectPartsDiskNotFound(obj ObjectLayer, instanceType string, fsDirs []string, t *testing.T) {
+
+	bucketNames := []string{"minio-bucket", "minio-2-bucket"}
+	objectNames := []string{"minio-object-1.txt"}
+	uploadIDs := []string{}
+
+	// bucketnames[0].
+	// objectNames[0].
+	// uploadIds [0].
+	// Create bucket before intiating NewMultipartUpload.
+	err := obj.MakeBucket(bucketNames[0])
+	if err != nil {
+		// Failed to create newbucket, abort.
+		t.Fatalf("%s : %s", instanceType, err.Error())
+	}
+	// Initiate Multipart Upload on the above created bucket.
+	uploadID, err := obj.NewMultipartUpload(bucketNames[0], objectNames[0], nil)
+	if err != nil {
+		// Failed to create NewMultipartUpload, abort.
+		t.Fatalf("%s : %s", instanceType, err.Error())
+	}
+
+	// Remove some random disk.
+	removeRoots([]string{fsDirs[len(fsDirs)-2]})
```

Comment:
Can we create a function in `test-utils_test.go` which then can be send as an argument for the test body and which can be called from inside. Than function could choose (n/2)-1 drives randomly by choosing numbers n between 1-n and bring them down. Later we can reuse that function.